src/inmanta/env.py

```diff
@@ -608,10 +617,11 @@ def check(cls, in_scope: Optional[Pattern[str]] = None, constraints: Optional[Li
             if (constraint.key not in installed_versions or str(installed_versions[constraint.key]) not in constraint)
             if not constraint.marker or constraint.marker.evaluate()
         ]
-        # constraints_for_marker = filter(lambda c: not (c.marker and not c.marker.evaluate()), constraints)
-        for constraint, v in constraint_violations:
-            LOGGER.warning("Incompatibility between constraint %s and installed version %s", constraint, v)
-        return len(constraint_violations) == 0
+        if len(constraint_violations) != 0:
+            msg: str = "Module dependency resolution conflict: a module dependency constraint \
+was violated by another module. This most likely indicates an incompatibility between \
+two or more of the installed modules."
```

Comment:
This is not in its place here. `ActiveEnv.check` does not know or care about modules, just the environment. This could be added higher up where the exception is caught.

Reply:
What should I put as message here then? As now it is not optional anymore? :p

Reply:
The same goes for me.